From 398f4d34b779c42576cf169811bf54d435e22520 Mon Sep 17 00:00:00 2001
From: Karl Williamson <khw@cpan.org>
Date: Wed, 24 Dec 2025 06:07:33 -0700
Subject: [PATCH 08/10] POSIX/t/time.t: tzset() works on Windows; not MingW

I ran some experiments, and found that tzset works on Windows, and is
required after changing the TZ environment variable from within perl.

But it did not work on MingW.  Maybe there is something else needed in
the Posix module that would get it to work; I didn't investigate

The only way I could figure out how to distinguish in Perl space between
MSVC and MingW was looking at the make command.  Maybe there is a better
way

(cherry picked from commit 220368d14b8434f7cc57d126021a719fe220b030)
---
 ext/POSIX/t/time.t | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/ext/POSIX/t/time.t b/ext/POSIX/t/time.t
index bccf6a6a12..694e03c942 100644
--- a/ext/POSIX/t/time.t
+++ b/ext/POSIX/t/time.t
@@ -17,11 +17,12 @@ use Test::More tests => 31;
 $ENV{TZ} = "EST5EDT";
 
 # It looks like POSIX.xs claims that only VMS and Mac OS traditional
-# don't have tzset().  Win32 works to call the function, but it doesn't
-# actually do anything.  Cygwin works in some places, but not others.  The
-# other Win32's below are guesses.
+# don't have tzset().  MingW doesn't work.  Cygwin works in some places, but
+# not others.  The other Win32's below are guesses.
 my $has_tzset = $^O ne "VMS" && $^O ne "cygwin"
-             && $^O ne "MSWin32" && $^O ne "interix";
+             && ($^O ne "MSWin32" || (   $^O eq "MSWin32"
+                                      && $Config{make} eq 'nmake'))
+             && $^O ne "interix";
 
 SKIP: {
     skip "No tzset()", 1 unless $has_tzset;
-- 
2.53.0

